### DISCUSSION TITLE
Git lfs bandwidth limit when public repo is cloned

### DISCUSSION THREAD

**Discussion by mark-andrews** [community/community#62051] [Repositories] [answered]
I want to make sure that I understand Git lfs bandwdith limits.
I have read this guide: https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-storage-and-bandwidth-usage

If I were to purchase a data pack currently @ $60 per year, that would give me 600GB bandwidth and 50GB storage per year. Is my bandwidth allowance used up when others clone the repo? I believe so. In the above guide it says **Forking and pulling a repository counts against the parent repository's bandwidth limit.**. But I just want to check that point here.

If I had a popular public repo, it seems like the allowed bandwidth could be consumed very quickly. For example, if I had one repo with 1GB, and 100 people cloned/forked/pulled it, then tha would use 100GB of *my*, not *their*, bandwidth?

With private repos, especially with small numbers of collaborators, it seems like we could be keep below the annual limit relatively easily. But we have no control over who clones/forks etc a public repo and so it seems that the bandwidth could be used up very quickly.

Am I understanding this correctly? Does this mean that public GitHub repos rarely use lfs?

**Comment by nirav-gajera** [comment, marked as answer]
Yes,  When someone clones, forks, or pulls a repository that uses Git LFS, the bandwidth usage is counted against the parent repository's bandwidth limit. This means that if you have a popular public repository with large files stored in Git LFS, your bandwidth limit could be consumed very quickly.

For example, if you have a public repository with 1GB of files stored in Git LFS, and 100 people clone or fork the repository, this would use 100GB of your bandwidth. This is because each clone or fork operation downloads the entire repository, including the large files that are stored in Git LFS.

Public GitHub repositories do rarely use Git LFS because of the bandwidth limits. If you want to use Git LFS for a public repository, you need to be aware of the bandwidth limits and make sure that you have enough bandwidth to support the number of clones, forks, and pulls that you expect to receive.

**Reply by mark-andrews** [reply]
Thanks a lot. It is very clear now.

**Comment by nnra6864** [comment]
Icl this has to be the dumbest design I've encountered in years.
So lets say I make a game that's 2-3GB in size, people decide to clone it or even grief my repo by constantly cloning, they can easily do so?
That makes 0 sense whatsoever...
